src/lib/comms/sol-lwm2m.c

```diff
@@ -2992,6 +2992,7 @@ handle_resource(struct sol_coap_server *server,
     r = extract_path(client, req, path, &path_size);
     header_code = SOL_COAP_RSPCODE_BAD_REQUEST;
     SOL_INT_CHECK_GOTO(r, < 0, exit);
+    SOL_INT_CHECK_GOTO(path_size, == 0, exit);
```

Comment:
if `path_size == 0` is not valid shouldn't the function `extract_path` return an error when this happen ?
Otherwise one will always need to do two checks, wheter return `< 0` or `path_size == 0`